### PR TITLE
8.0 tokudb

### PR DESCRIPTION
**Pull request by georgelorchpercona**

### CODE REVIEW

**Comment by georgelorchpercona** [comment]
https://jenkins.percona.com/view/TokuDB/job/PerconaFT-param/119/
https://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/2135/
https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1744/

File: cmake_modules/TokuSetupCompiler.cmake [hunk 1]
```diff
@@ -142,6 +142,11 @@ set_ldflags_if_supported(
   -Wno-error=strict-overflow
   )
 
+# new flag sets in MySQL 8.0 seem to explicitly disable this
+set_cflags_if_supported(
```
Comment (laurynas-biveinis) [inline cmake_modules/TokuSetupCompiler.cmake]: Just set without checking if supported? It's a very old flag

File: portability/memory.cc [hunk 2]
```diff
@@ -176,7 +176,7 @@ toku_memory_footprint(void * p, size_t touched)
 
 void *
 toku_malloc(size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/memory.cc [hunk 3]
```diff
@@ -203,7 +203,7 @@ toku_malloc(size_t size) {
 }
 
 void *toku_malloc_aligned(size_t alignment, size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/memory.cc [hunk 4]
```diff
@@ -239,7 +239,7 @@ toku_calloc(size_t nmemb, size_t size) {
 
 void *
 toku_realloc(void *p, size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/memory.cc [hunk 5]
```diff
@@ -270,7 +270,7 @@ toku_realloc(void *p, size_t size) {
 }
 
 void *toku_realloc_aligned(size_t alignment, void *p, size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/memory.cc [hunk 6]
```diff
@@ -339,7 +339,7 @@ toku_free(void *p) {
 
 void *
 toku_xmalloc(size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/memory.cc [hunk 7]
```diff
@@ -369,7 +369,7 @@ void* toku_xmalloc_aligned(size_t alignment, size_t size)
 //  Fail with a resource_assert if the allocation fails (don't return an error code).
 // Requires: alignment is a power of two.
 {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/memory.cc [hunk 8]
```diff
@@ -403,7 +403,7 @@ toku_xcalloc(size_t nmemb, size_t size) {
 
 void *
 toku_xrealloc(void *v, size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```
Comment (laurynas-biveinis) [inline portability/memory.cc]: just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)

File: portability/toku_portability.h [hunk 9]
```diff
@@ -69,7 +69,7 @@ Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
 #include <sys/stat.h>
 #include <stdio.h>
 
-#if __FreeBSD__
+#if defined(__FreeBSD__) && __FreeBSD__
```
Comment (laurynas-biveinis) [inline portability/toku_portability.h]: Just if defined or ifdef, no need to check the actual value according to https://www.freebsd.org/doc/en/books/porters-handbook/porting-versions.html

**Comment by georgelorchpercona** [comment]
Done. Compiled and tested 5.6, 5.7, and 8.0 locally, no jenkins as I don't think it is needed for these changes.

**Pull request merged**